### PR TITLE
fix postLogin and postLogout hooks

### PR DESCRIPTION
**Pull request by fiedsch**
both hooks apply to backend and front end users as they are triggered in the  `User` class that both `FrontendUser`and `BackendUser` are derived from.

### CODE REVIEW

**Comment by Toflar** [comment]
Good find! I think rather than checking for `TL_MODE == 'FE'` you should check for `$user instanceof \FrontendUser` :) We should discourage using constants wherever we can :)

**Comment by fiedsch** [comment]
Thanks for the comment, Yanick. I updated the PR. 

I came across the problem by https://community.contao.org/de/showthread.php?60122-postLogin-BackendUser-given&p=391129

**Comment by Toflar** [comment]
:+1:

File: api/extensions/hooks/postLogin.md [hunk 1]
```diff
@@ -1,16 +1,17 @@
 # postLogin
 
-The `postLogin` hook is triggered after a user has logged into the front end. It
-passes the user object as argument and does not expect a return value.
+The `postLogin` hook is triggered after a user has logged in. This can 
+be either in the back end or the fornt end. It passes the user object
```
Comment (aschempp) [inline api/extensions/hooks/postLogin.md]: fornt => front :see_no_evil:

File: api/extensions/hooks/postLogin.md [hunk 2]
```diff
@@ -22,9 +23,11 @@ passes the user object as argument and does not expect a return value.
 $GLOBALS['TL_HOOKS']['postLogin'][] = array('MyClass', 'myPostLogin');
 
 // MyClass.php
-public function myPostLogin(FrontendUser $objUser)
+public function myPostLogin(User $user)
 {
-    // Do something
+    if ($user instanceof FrontendUser) {
+     // Do something with the for front end user $user
```
Comment (aschempp) [inline api/extensions/hooks/postLogin.md]: Remove "for"

File: api/extensions/hooks/postLogout.md [hunk 3]
```diff
@@ -22,9 +22,11 @@ end. It passes the user object as argument and does not expect a return value.
 $GLOBALS['TL_HOOKS']['postLogout'][] = array('MyClass', 'myPostLogout');
 
 // MyClass.php
-public function myPostLogout(FrontendUser $objUser)
+public function myPostLogout(User $user)
 {
-    // Do something
+    if ($user instanceof FrontendUser) {
+     // Do something with the for front end user $user
```
Comment (aschempp) [inline api/extensions/hooks/postLogout.md]: Please always use four white spaces for indentation

**Comment by fiedsch** [comment]
Thanks for carefull reading. Hope to create a PR without issues of that kind sometime soon. 
The PR is updated.

**Comment by aschempp** [comment]
No problem at all! That's the power of open source, multiple people can contribute for the best result :-)

**Pull request merged**